src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java

```diff
@@ -137,21 +127,12 @@ public void add(JRule jRule, boolean enableRule) {
     }
 
     public void addDynamicWhenReceivedCommand(Method method, JRule jRule, String ruleName, List<String> items) {
-        JRuleModuleEntry ruleModuleEntry = new JRuleModuleEntry(jRule, method, ruleName);
-        List<JRulePreconditionContext> emptyPreconditionContextList = new ArrayList<>(0);
+        JRuleBuilder jRuleBuilder = createJRuleBuilder(ruleName, jRule, method);
         for (String itemName : items) {
-            JRuleItemReceivedCommandExecutionContext context = new JRuleItemReceivedCommandExecutionContext(jRule,
-                    ruleName, JRuleEngine.EMPTY_LOG_TAGS, method, itemName, JRuleMemberOf.None, Optional.empty(),
-                    emptyPreconditionContextList, Optional.empty(), null, null);
-
-            addToContext(context, false);
-            ruleLoadingStatistics.addItemStateTrigger();
-            ruleModuleEntry.addJRuleWhenItemReceivedCommand(context);
-
+            jRuleBuilder.whenItemReceivedCommand(itemName, JRuleMemberOf.None, null, null);
             logInfo("Adding Dynamic Rule Name: {} item: {}", ruleName, itemName);
         }
-        ruleLoadingStatistics.addRuleClass();
```

Comment:
Where did the addRuleClass go? It is removed here and not in JRuleBuilder

Reply:
You are totally right. that call somehow got lost. I added it again in the last commit.